neqo-http3/src/connection_client.rs

```diff
@@ -4404,6 +4458,30 @@ mod tests {
         assert_eq!(client.cancel_push(0), Err(Error::InvalidStreamId));
     }
 
+    /// We can't keep the connection alive on the basis of a push promise,
+    /// nor do we want to if the push promise is not interesting to the client.
+    /// We do the next best thing, which is keep any push stream alive if the
+    /// client reads from it.
+    #[test]
```

Comment:
We also need a test that  push_promise will no cause keep live to be active. Probably also that the keep alive is not active if client does not call read_push_data.

Reply:
Yeah, I think that I can manage that in the same test: promise push, send response, read response, client is idle, send (incomplete) push data, read push data, client is not idle